spring-core/src/main/java/org/togglz/spring/activation/SpringEnvironmentPropertyActivationStrategy.java

```diff
@@ -0,0 +1,71 @@
+package org.togglz.spring.activation;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+import org.togglz.core.Feature;
+import org.togglz.core.activation.Parameter;
+import org.togglz.core.activation.ParameterBuilder;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.user.FeatureUser;
+import org.togglz.core.util.Strings;
+import org.togglz.spring.util.ContextClassLoaderApplicationContextHolder;
+
+/**
+ * <p>
+ * An activation strategy based on the values of properties accessible within the Spring environment.
+ * </p>
+ * <p>
+ * It can either be based on a given property name, passed as a parameter, or a property name constructed from the
+ * {@link Feature} itself (e.g. {@code com.example.MyFeatures.FEATURE_NAME}).
+ * </p>
+ *
+ * @author Alasdair Mercer
+ */
+public class SpringEnvironmentPropertyActivationStrategy implements ActivationStrategy {
+
+    public static final String ID = "spring-environment-property";
```

Comment:
I was simply following the standard of having ID and PARAM constants public so that they can be used by consumers of the API, presumably to avoid them from hard-coding these values and potentially breaking on upgrades.